pwiz_tools/Skyline/Model/AlphaPeptDeep/AlphapeptdeepLibraryBuilder.cs

```diff
@@ -0,0 +1,463 @@
+                { @"FragmentNumber", @"FragmentSeriesNumber" },
+                { CCS, COLLISIONAL_CROSS_SECTION},
+                { ION_MOBILITY, null}
+            };
+
+        public AlphapeptdeepLibraryBuilder(string libName, string libOutPath, string pythonVirtualEnvironmentScriptsDir,
+            SrmDocument document, IrtStandard irtStandard)
+
+        {
+            Document = document;
+            IrtStandard = irtStandard;
+            LibrarySpec = new BiblioSpecLiteSpec(libName, libOutPath);
+            PythonVirtualEnvironmentScriptsDir = pythonVirtualEnvironmentScriptsDir;
+
+            ToolName = @"AlphaPeptDeep";
+
+            if (RootDir == null)
+            {
+                RootDir = Path.GetDirectoryName(libOutPath);
+            }
+
+            if (RootDir != null)
+            {
+                RootDir = Path.Combine(RootDir, libName);
+                //Directory.CreateDirectory(RootDir);
+                InitializeLibraryHelper(RootDir);
+            }
+        }
+
+        private void InitializeLibraryHelper(string rootDir)
+        {
+            if (LibraryHelper == null)
+            {
+                LibraryHelper = new LibraryHelper(rootDir, ALPHAPEPTDEEP);
+                RootDir = LibraryHelper.GetRootDir(rootDir, ALPHAPEPTDEEP);
+                LibraryHelper.InitializeLibraryHelper(InputFilePath);
+            }
+        }
+        public bool BuildLibrary(IProgressMonitor progress)
+        {
+            IProgressStatus progressStatus = new ProgressStatus();
+
+            try
+            {
+                InitializeLibraryHelper(RootDir);
+                RunAlphapeptdeep(progress, ref progressStatus);
+                progress.UpdateProgress(progressStatus = progressStatus.Complete());
+                LibraryHelper = null;
+                return true;
+            }
+            catch (Exception exception)
+            {
+                progress.UpdateProgress(progressStatus.ChangeErrorException(exception));
+                LibraryHelper = null;
+                return false;
+            }
+        }
+
+        private void RunAlphapeptdeep(IProgressMonitor progress, ref IProgressStatus progressStatus)
+        {
+            // DSHTEYN:  These should be better balanced as of May 2nd 2025
+            var segmentEndPercentages = new[] { 5, 10, 85, 100 };
+            progressStatus = progressStatus.ChangeSegments(0, ImmutableList<int>.ValueOf( segmentEndPercentages));
+            LibraryHelper.PreparePrecursorInputFile(Document, progress, ref progressStatus, @"AlphaPeptDeep",
+                IrtStandard);
+            progressStatus = progressStatus.NextSegment();
+            PrepareSettingsFile(progress, ref progressStatus);
+            ExecutePeptdeep(progress, ref progressStatus);
+            TransformPeptdeepOutput(progress, ref progressStatus);
+            progressStatus = progressStatus.NextSegment();
+            ImportSpectralLibrary(progress, ref progressStatus);
```

Comment:
I think running BlibBuild.exe only works if the progressStatus that you pass into it is on the last segment.
The problem is that in BlibBuild.BuildLibrary, it is using the following to decide whether it successfully completed:
`isComplete = status.IsComplete;`
If "isComplete" ends up being false, it deletes the .blib file it just created.